### PR TITLE
feat(material/chips): allow custom default color

### PR DESCRIPTION
**Pull request by Nxtivision**
Lets the user choose what color will be the default instead of 'primary' via mixinColor

✅ Tests adapted
✅ No breaking change: I keep the same default color `primary`

### CODE REVIEW

File: src/material/chips/chip.ts [hunk 1]
```diff
@@ -236,8 +240,12 @@ export class MatChip
       this.tabIndex = parseInt(tabIndex) ?? this.defaultTabIndex;
     }
     this._monitorFocus();
+    this.color = _defaultOptions?.color;
   }
 
+  /** Action receiving the primary set of user interactions. */
```
Comment (andrewseguin) [inline src/material/chips/chip.ts:246]: Can you move this back above the constructor?

File: src/material/chips/chip.spec.ts [hunk 2]
```diff
@@ -215,6 +253,15 @@ class SingleChip {
   chipRemove: (event?: MatChipEvent) => void = () => {};
 }
 
+@Component({
```
Comment (andrewseguin) [inline src/material/chips/chip.spec.ts:256]: You can use the existing `BasicChip` for your test instead

File: src/material/chips/chip.spec.ts [hunk 3]
```diff
@@ -127,6 +138,32 @@ describe('MDC-based MatChip', () => {
       expect(chipNativeElement.classList).toContain('mat-warn');
     });
 
+    it('can default the color via MAT_CHIPS_DEFAULT_OPTIONS provider', () => {
```
Comment (andrewseguin) [inline src/material/chips/chip.spec.ts:141]: Move this to the `describe('MatBasicChip`)` test suite

File: src/material/chips/chip.spec.ts [hunk 4]
```diff
@@ -215,6 +250,15 @@ class SingleChip {
   chipRemove: (event?: MatChipEvent) => void = () => {};
 }
 
+@Component({
```
Comment (andrewseguin) [inline src/material/chips/chip.spec.ts:253]: This can be removed

**Review by andrewseguin — APPROVED** [review]

**Comment by Nxtivision** [comment]
Thank you! 😄

**Comment by andrewseguin** [comment]
Thank you for the contribution :) I'll run this by internal Google tests to make sure it can be merged

**Comment by andrewseguin** [comment]
One more thing - youll need to run ` yarn bazel run //tools/public_api_guard:material/chips.md_api.accept` to edit the API goldens

**Comment by andrewseguin** [comment]
There were some failures internally, I'll try to take a look this week to determine what's going on

**Comment by andrewseguin** [comment]
Hey thanks for your patience with this PR - this was on the backburner as we prepare v15's MDC launch. I looked at the failures. We need just one small change to `chip-row.ts` which calls the super `MatChip` constructor. You'll want to inject and pass the default options in the constructor.

**Comment by andrewseguin** [comment]
Good news is that now all tests pass internally and externally, but unfortunately this missed our window for getting in for v15. We're in a feature freeze until November's release, but once that passes we can get this merged in. Thanks for your contribution and patience :)

**Comment by Nxtivision** [comment]
Hi Andrew,
Thank you for letting me know about this ! :) Can we expect it to be released before the end of this year ?

**Comment by andrewseguin** [comment]
Sorry this never got in. We're going to close this since the `color` API is not something Material is designing for anymore. In M3, we recommend clients just apply different color themes using CSS

**Pull request closed without merging**